docs/reference/action_list.rst

```diff
@@ -226,6 +226,115 @@ to more field types, see `SonataDoctrineORMAdminBundle Documentation`_.
     values so use the ``inverse`` option if you really cannot find a good enough
     antonym for the name you have.
 
+Symfony Data Transformers
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If the model field has a limited list of values (enumeration), it is convenient to use a value object to control
+the available values. For example, consider the value object of moderation status with the following values:
+``awaiting``, ``approved``, ``rejected``::
+
+    final class ModerationStatus
+    {
+        public const AWAITING = 'awaiting';
+        public const APPROVED = 'approved';
+        public const REJECTED = 'rejected';
+
+        private static $instances = [];
+
+        private string $value;
+
+        private function __construct(string $value)
+        {
+            if (!array_key_exists($value, self::choices())) {
+                throw new \DomainException(sprintf('The value "%s" is not a valid moderation status.', $value));
+            }
+
+            $this->value = $value;
+        }
+
+        public static function byValue(string $value): Permission
```

Comment:
I guess the return declaration is wrong in this example.

Reply:
Sorry for this. Fixed